test/integration/targets/loops/tasks/main.yml

```diff
@@ -376,35 +376,50 @@
     that:
       - loop_out['results'][1]['ansible_remote_tmp'] == '/tmp/test1'
 
-# https://github.com/ansible/ansible/issues/64169
-- include_vars: 64169.yml
-
-- set_fact: "{{ item.key }}={{ hostvars[inventory_hostname][item.value] }}"
-  with_dict:
-    foo: __foo
-
-- debug:
-    var: foo
-
-- assert:
-    that:
-      - foo[0] != 'foo1.0'
-      - foo[0] == unsafe_value
+- name: Ensure we only flag unsafe with_ loops (due to lookups) https://github.com/ansible/ansible/issues/64169
+  tags: unsafe
   vars:
+    # version_64169 set in vars/main.yml
+    version_64169: '1.0'
     unsafe_value: !unsafe 'foo{{ version_64169 }}'
+  block:
+  - name: include __foo with a templatable item
+    include_vars: 64169.yml
 
-- set_fact: "{{ item.key }}={{ hostvars[inventory_hostname][item.value] }}"
-  loop: "{{ dicty_dict|dict2items }}"
-  vars:
-    dicty_dict:
+  - name: Create fact using with_dict (akak lookup('dict')), should be unsafe
+    set_fact: "{{ item.key }}={{ hostvars[inventory_hostname][item.value] }}"
+    with_dict:
       foo: __foo
 
-- debug:
-    var: foo
-
-- assert:
-    that:
-      - foo[0] == 'foo1.0'
+  - debug:
+      var: foo
+    when: ansible_verbosity >= 1
+
+  - name: Ensure result is unsafe
+    assert:
+      that:
+        - foo[0] | type_debug == 'AnsibleUnsafeText'
+
+        - foo[0] != 'foo1.0'
+        - foo[0] == unsafe_value
+
+  - name: Create fact using using normal loop, should not be unsafe
```

Comment:
I don't get it, why should this not be unsafe? We are still loading a variable from an unsafe location which should taint the fact being set. What's the difference between `with_*` and `loop` here? It seems like this test is not something we should have supported in the first place.

Reply:
loops are not unsafe by nature, the reason with_ is flagged unsafe is that  lookups are considered unsafe. If a loop uses a lookup, normal templating will already flag as unsafe.